sklearn/linear_model/ridge.py

```diff
@@ -141,7 +145,17 @@ def _solve_cholesky_kernel(K, y, alpha, sample_weight=None):
         # Only one penalty, we can solve multi-target problems in one time.
         K.flat[::n_samples + 1] += alpha[0]
 
-        dual_coef = linalg.solve(K, y, sym_pos=True, overwrite_a=True)
+        try:
+            # Note: we must use overwrite_a=False in order to be able to
+            #       use the fall-back solution below in case a LinAlgError
+            #       is raised
+            dual_coef = linalg.solve(K, y, sym_pos=True,
+                                     overwrite_a=False)
+        except np.linalg.LinAlgError:
+            warnings.warn("Singular matrix in solving dual problem. Using "
+                          "pseudo inverse instead.")
+            Kinv = linalg.pinv2(K)
+            dual_coef = np.dot(Kinv, y)
```

Comment:
From the history, @eickenberg also commented the following:
np.linalg.lstsq accepts multiple targets. However, afaik it does an SVD. So that would be almost as costly as the full pseudoinverse. I am starting to understand :) But I'd still prefer a least squares solver.